### DISCUSSION TITLE
UX/UI: Transaction history / details + menu display

### DISCUSSION THREAD

**Discussion by swedishfrenchpress** [cashubtc/eNuts#224] [Announcements]
One concern I have is the color choices of green and red for the incoming and outgoing satoshi amounts in the transaction history. Given that red/green color blindness is the most prevalent, using this color combination without a "+" or "-" indicator could be confusing for users. This indicator would clearly signify the direction of funds, making it best practice.

I also have feedback on the display of line items in the settings. The current UI encloses the text in ovals, which seem to add unnecessary visual elements. Some suggestions:

**Transaction History**
![s1](https://github.com/cashubtc/eNuts/assets/78821053/abfd2d6e-1b63-46d5-8b13-c506308ddcdd)
- Streamline the settings display with underlined rows.
- Use left and right arrows to indicate incoming and outgoing transactions.
- Add "+" and "-" to transactions for clarity on direction.
- Displaying dates under each transaction seems redundant; consider grouping transactions under a date header.
- Use a space, not a comma, as the thousands separator for better legibility.
- Replaced thunderbolt icon with sats denomination

**Payment Details**
![s2](https://github.com/cashubtc/eNuts/assets/78821053/885cb9a6-41dc-4cf8-a15d-0bfebc35a1c4)
- Add a "Transaction Details" header.
- Remove the comma between the date and time.
- Renames satoshi to sats

**Miscellaneous Thoughts**
I'm open to all any feedback here. I know this design is suggestion a pretty large change from the existing UI and the last thing I want do is create extra work for the sake of work, but I do believe that some of these suggestions would help.

**Comment by swedishfrenchpress** [comment]
Exploring a version with 15px icons.
![S5](https://github.com/cashubtc/eNuts/assets/78821053/06bdb7ea-cb8a-4741-92cb-03da60953bc9)

**Comment by KKA11010** [comment]
Converting this into a discussion

**Comment by swedishfrenchpress** [comment]
If we are planning to modify the menu styling for transactions, nostr contact list, and so on, then I believe it is needed to update the styling for consistency across various sections of the screen. Other areas of the app, such as the disclaimer and settings, will also benefit from styling updates. For example:
![7](https://github.com/cashubtc/eNuts/assets/78821053/73841841-3a9a-4702-9275-3025ce2b66f4)

**Comment by swedishfrenchpress** [comment]
Some continuous iteration here. I think this design does a better job of separating the sections and increasing contrast between the menu options and the background (something you did very well in your original design. It may require some styling changes across the application though. Would love to know what you think.

![UI Changes](https://github.com/cashubtc/eNuts/assets/78821053/b859afec-b974-4330-841e-4a13316c795e)
![enuts-side-by-side](https://github.com/cashubtc/eNuts/assets/78821053/2aae40c5-bc2d-46f6-9570-252cf9475825)

**Reply by KKA11010** [reply]
Thank you very much for this proposals. I prefer A and I have already done a bit regarding the screens designs. Please run the branch nostr-cache to see the dark theme changes.

**Reply by KKA11010** [reply]
the branch is a work in progress so dont be surprised if the nostr intergration still dont work as expected

**Reply by KKA11010** [reply]
@swedishfrenchpress Actually I like both and its hard for me to decide :) but we should make a decision asap so we can mark the dark theme design as done and move forward. After thinking more about it, I would prefer B even tho it introduces more changes to the current design.

**Reply by swedishfrenchpress** [reply]
I've had time to sit on it and get some feedback during the hackday. B is the clear winner and I think it's the way to go for menu items / input fields going forward. I've updated the "production ready" Figma page to match this style.
